Exercise_1.java

```diff
@@ -1,8 +1,29 @@
+// Time Complexity : O(log(N))
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode :
+// Any problem you faced while coding this : No
+
+
 class BinarySearch { 
     // Returns index of x if it is present in arr[l.. r], else return -1 
     int binarySearch(int arr[], int l, int r, int x) 
     { 
         //Write your code here
+        int index = -1;    //default index is -1 as it represents that element is not found.
+        while(l <= r){
+            int mid = (l+r)/2;   //searching only the satisfied space as the array is sorted.

```

Comment:
this condition will fail for numbers approaching max and min values of int
use mid = l + (r-l)/2